cwms_radar_api/src/main/java/cwms/radar/api/LocationController.java

```diff
@@ -352,21 +369,19 @@ public void update(Context ctx, @NotNull String locationId) {
     @Override
     public void delete(Context ctx, @NotNull String locationId) {
 
-        try (final Timer.Context timeContext = markAndTime(DELETE);
+        String office = ctx.queryParam(OFFICE);
+        try (Timer.Context timeContext = markAndTime(DELETE);
              DSLContext dsl = getDslContext(ctx)) {
-            String office = ctx.queryParam(OFFICE);
             LocationsDao locationsDao = getLocationsDao(dsl);
-            locationsDao.deleteLocation(locationId, office);
+            boolean cascadeDelete = ctx.queryParamAsClass(CASCADE_DELETE, Boolean.class).getOrDefault(false);
+            locationsDao.deleteLocation(locationId, office, cascadeDelete);
             ctx.status(HttpServletResponse.SC_ACCEPTED).json(locationId + " Deleted");
-        } catch (NotFoundException e) {
-            RadarError re = new RadarError("Not found.");
-            logger.log(Level.WARNING, re.toString(), e);
-            ctx.status(HttpServletResponse.SC_NOT_FOUND);
-            ctx.json(re);
-        } catch (IOException ex) {
-            RadarError re = new RadarError("Failed to delete location");
-            logger.log(Level.SEVERE, re.toString(), ex);
-            ctx.status(HttpServletResponse.SC_INTERNAL_SERVER_ERROR).json(re);
+        } catch (DataAccessException ex) {
+            SQLException cause = ex.getCause(SQLException.class);
+            if(cause != null && cause.getErrorCode() == 20031) {
```

Comment:
error code 20031 is specific to deleting locations so I didn't put this up in the general ApiServlet handling

Reply:
Makes sense. though it might be worth organizing all of those db error codes into a specific class at some point.

Reply:
a mechanism for codgen would be nice

Reply:
Honestly wouldn't be too hard. or really even need codegen (if you meant JOOQ), the errors are already listed as rows in a table.

Reply:
Yeah I was thinking of generating enums for some of the CWMS owned static tables: https://stackoverflow.com/questions/46429671/generate-enum-class-from-table-with-jooq

Reply:
Alternatively that code could get moved to the wrapException handler. But in this case this does seem pretty specific to this usage.